sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubClient.java

```diff
@@ -232,92 +301,109 @@ public IncomingMessage(
   }
 
   /**
+   * Create a client using the underlying transport.
+   */
+  public static PubsubClient newClient(
+      TransportType transportType,
+      @Nullable String timestampLabel,
+      @Nullable String idLabel,
+      PubsubOptions options) throws IOException {
+    switch (transportType) {
+      case APIARY:
+        return PubsubApiaryClient.newClient(timestampLabel, idLabel, options);
+      case GRPC:
+        return PubsubGrpcClient.newClient(timestampLabel, idLabel, options);
+    }
+    throw new RuntimeException(); // cases are exhaustive.
+  }
+
+  /**
    * Gracefully close the underlying transport.
    */
   @Override
-  void close();
-
+  public abstract void close();
```

Comment:
can you make it `Closeable` instead of `AutoCloseable`? Stronger guarantees.

Do you need to override the Javadoc comment? If so, maybe also add `@inheritDoc` or something to link to parent interface